modmail/log.py

```diff
@@ -1,7 +1,105 @@
 import logging
+import pathlib
 from typing import Any
 
 
+__all__ = [
+    "DEFAULT",
+    "get_logging_level",
+    "set_logger_levels",
+    "ModmailLogger",
+]
+
+logging.TRACE = 5
+logging.NOTICE = 25
+logging.addLevelName(logging.TRACE, "TRACE")
+logging.addLevelName(logging.NOTICE, "NOTICE")
+
+DEFAULT = logging.INFO
+
+
+def _get_env() -> dict:
+    import os
+
+    try:
+        from dotenv import dotenv_values
+    except ModuleNotFoundError:
+        dotenv_values = lambda *args, **kwargs: dict()  # noqa: E731
+
+    return {**dotenv_values(), **os.environ}
+
+
+def get_logging_level() -> None:
+    """Get the configured logging level, defaulting to logging.INFO."""
+    key = "MODMAIL_LOG_LEVEL"
+
+    level = _get_env().get(key, DEFAULT)
+
+    try:
+        level = int(level)
+    except TypeError:
+        level = DEFAULT
+    except ValueError:
+        level = level.upper()
+        if hasattr(logging, level) and isinstance(getattr(logging, level), int):
+            return getattr(logging, level)
+        print(
+            f"Environment variable {key} must be able to be converted into an integer.\n"
+            f"To resolve this issue, set {key} to an integer value, or remove it from the environment.\n"
+            "It is also possible that it is sourced from an .env file."
+        )
+        exit(1)
+
+    return level
+
+
+def set_logger_levels() -> None:
+    """
+    Set all loggers to the provided environment variables.
+
+    eg MODMAIL_TRACE_LOGGERS will be split by `,` and each logger will be set to the trace level
+    This is applied for every logging level.
+    """
+    env_vars = _get_env()
+    fmt_key = "MODMAIL_{level}_LOGGERS"
```

Comment:
TBH I'm not sure if this should be this format, or if `MODMAIL_LOGGERS_{level}` makes more sense since the variable is now more hierarchical.

Reply:
I'm going to reverse this format, actually.